src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -204,13 +204,8 @@ export function uiRenderMixin(kbnServer, server, config) {
     async handler(req, h) {
       const id = req.params.id;
       const app = server.getUiAppById(id);
-
       try {
-        if (kbnServer.status.isGreen()) {
-          return await h.renderApp(app);
-        } else {
-          return await h.renderStatusPage();
-        }
+        return await h.renderApp(app);
```

Comment:
This block was mostly broken since we moved the ES service and status check to KP, as we are awaiting for ES to be ready before starting up the plugins. Only situation it would properly work is if the server status would change from green to red after initialization. 
As this part (auto 503 handling) is another step of https://github.com/elastic/kibana/issues/41983, and as we can't easily render the status page from legacy `h` helper anyway, I think this change is alright for now.

Reply:
> This block was mostly broken since we moved the ES service and status check to KP, as we are awaiting for ES to be ready before starting up the plugins. 

we could serve `status` page with `NotReadyServer` while waiting for ES service to be ready
https://github.com/elastic/kibana/blob/9e8448fc0604474eb4de73f5c899584fb3055bac/src/core/server/http/http_service.ts#L187-L201

That's optional suggestion and not a blocker for the current PR

Reply:
Yea, we do want to do that, however this is far from trivial from what I saw:

We are going to have a cyclic dependency issue doing so, as this `notReadyServer`  is within the http service, and we gonna need to use the `httpResources`, or at least the `rendering` service to send the correct response. We could expose something like an internal `registerNotReadyRoute` or something to work-around this issue.

Second point, this will not work until we migrate the server-side status endpoints to core, as atm the endpoints needed by the page are in a legacy plugin, so the routes are not mounted until the end of core's start phase. Even when migrated, we would need to register these routes to the `notReadyServer` too, for them to be available during the setup phase.

Follow-up of previous point, the `notReadyServer` is currently a 'raw' HAPI server, so not directly compatible with our route registrations API. We would need to adapt the server to be compatible, or adapt the routes.

I added a point in https://github.com/elastic/kibana/issues/41983 and linked it to this discussion.